dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderFactoryImpl.java

```diff
@@ -223,13 +205,18 @@ protected Folder findFolderByPath(String path, Host host) throws DotDataExceptio
 					hostId = host.getIdentifier();
 				}
 
-				HibernateUtil dh = new HibernateUtil(Folder.class);
-				dh.setSQLQuery("select {folder.*} from folder, inode folder_1_, identifier identifier where asset_name = ? and parent_path = ? and "
+				DotConnect dc = new DotConnect();
+				dc.setSQL("select folder.* from " + Type.FOLDER.getTableName() + " folder, inode folder_1_, identifier  where asset_name = ? and parent_path = ? and "
 						+ "folder_1_.type = 'folder' and folder.inode = folder_1_.inode and folder.identifier = identifier.id and host_inode = ?");
-				dh.setParam(assetName.toLowerCase());
-				dh.setParam(parentPath.toLowerCase());
-				dh.setParam(hostId);
-				folder = (Folder) dh.load();
+				dc.addParam(assetName.toLowerCase());
+				dc.addParam(parentPath.toLowerCase());
+				dc.addParam(hostId);
+
+				result = ConvertToPOJOUtil.convertDotConnectMapToFolder(dc.loadResults());
+
+				if (result != null && !result.isEmpty()){
+					folder = result.get(0);
+				}
 
 				// if it is found add it to folder cache
 				if(UtilMethods.isSet(folder) && UtilMethods.isSet(folder.getInode())) {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "folder" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)